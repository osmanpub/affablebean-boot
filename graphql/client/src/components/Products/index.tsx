import { useQuery } from "@apollo/react-hooks";
import React from "react";
import { connect, useDispatch } from "react-redux";
import { Link } from "react-router-dom";
import { getId } from "../../helpers/utils";
import { Cart } from "../../interfaces/cart";
import { CategoryState, ProductState } from "../../interfaces/categories";
import { ID } from "../../interfaces/id";
import { addProductToCart, updateProductInCart } from "../../net/cart";
import { GET_CATEGORY_PRODUCTS } from "../../graphql/queries";
import { RootState } from "../../redux";
import { clearPurchase } from "../../redux/purchase";
import "./Products.css";
import {
  CategoryTitle,
  ProductsLeft,
  ProductsRight,
  ProductsTable,
  SelectedCategory,
} from "./Products.styles";

type Props = {
  cart: Cart;
  clearPurchase: Function;
  id: string;
};

function Products(props: Props) {
  const { cart, clearPurchase, id } = props;
  const { data } = useQuery(GET_CATEGORY_PRODUCTS, { variables: { id } });
  const dispatch = useDispatch();

  if (!data || !data.category) {
    return null;
  }

  const { categories, category, products } = data.category;

  if (
    !categories ||
    !categories.length ||
    !category ||
    !products ||
    !products.length
  ) {
    return null;
  }

  const addToCart = (id: ID) => {
    const update = cart.items.filter((item) => getId(item.product) === id);

    clearPurchase();
    dispatch(
      update.length > 0
        ? updateProductInCart(id, update[0].quantity + 1)
        : addProductToCart(id.toString())
    );
  };

  if (categories.length === 0) {
    return null;
  }

  const selectedCategory = category;

  const sidePanel = categories.map((category: CategoryState) => {
    const id = getId(category);
    const name = category.name;

    if (name === selectedCategory.name) {
      return (
        <SelectedCategory key={id} data-testid={`selected-${name}`}>
          <span className="categoryText">{name}</span>
        </SelectedCategory>
      );
    } else {
      return (
        <Link key={id} to={`/category/${id}`}>
          <span className="categoryButton">
            <span
              className="categoryText"
              data-cy={`category-${name}`}
              data-testid={`category-${name}`}
            >
              {name}
            </span>
          </span>
        </Link>
      );
    }
  });

  const productsList = products.map((product: ProductState, index: number) => {
    const id = getId(product);
    const name = product.name;
    const rowCol = index % 2 === 0 ? "white" : "lightBlue";

    return (
      <tr
        key={id}
        className={`${rowCol}`}
        data-cy={`product-${name}`}
        data-testid="product"
      >
        <td>
          <img src={`/static/img/products/${name}.png`} alt={name} />
        </td>

        <td data-testid={name}>
          {name}
          <br />
          <span className="smallText">{product.description}</span>
        </td>

        <td data-testid={`price-${name}`}>
          &euro;&nbsp;
          {product.price.toFixed(2)}
        </td>

        <td>&nbsp;&nbsp;</td>

        <td>
          <button
            className="btn btn-primary btn-sm"
            data-testid={`add-qty-${name}`}
            onClick={() => addToCart(id)}
          >
            add
          </button>
        </td>

        <td>&nbsp;&nbsp;</td>
      </tr>
    );
  });

  return (
    <div>
      <ProductsLeft>{sidePanel}</ProductsLeft>
      <ProductsRight>
        <CategoryTitle>{selectedCategory.name}</CategoryTitle>
        <ProductsTable>
          <tbody>{productsList}</tbody>
        </ProductsTable>
      </ProductsRight>
    </div>
  );
}

const mapStateToProps = (state: RootState) => ({
  cart: state.cart,
});

const mapDispatchToProps = {
  clearPurchase,
};

export default connect(mapStateToProps, mapDispatchToProps)(Products);
