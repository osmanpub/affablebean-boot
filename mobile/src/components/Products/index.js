import React, {Component} from 'react';
import PropTypes from 'prop-types';
import {Button, FlatList, Image, Text, View} from 'react-native';
import {addProductToCart, updateProductInCart} from '../../net/cart';
import {clearPurchase} from '../../actions';

export default class Products extends Component {
  addToCart = id => {
    const {cart, dispatch} = this.props;
    const update = cart.items.filter(item => item.product.id === id);

    dispatch(clearPurchase());
    dispatch(
      update.length > 0
        ? updateProductInCart(id, update[0].quantity + 1)
        : addProductToCart(id),
    );
  };

  getProductIcon = name => {
    var icon = '';

    switch (name) {
      case 'broccoli':
        icon = require('../../../assets/img/products/broccoli.png');
        break;

      case 'butter':
        icon = require('../../../assets/img/products/butter.png');
        break;

      case 'cheese':
        icon = require('../../../assets/img/products/cheese.png');
        break;

      case 'chicken leg':
        icon = require('../../../assets/img/products/chicken-leg.png');
        break;

      case 'chocolate cookies':
        icon = require('../../../assets/img/products/chocolate-cookies.png');
        break;

      case 'corn on the cob':
        icon = require('../../../assets/img/products/corn-on-the-cob.png');
        break;

      case 'free range eggs':
        icon = require('../../../assets/img/products/free-range-eggs.png');
        break;

      case 'granola':
        icon = require('../../../assets/img/products/granola.png');
        break;

      case 'green tea':
        icon = require('../../../assets/img/products/green-tea.png');
        break;

      case 'herbal tea':
        icon = require('../../../assets/img/products/herbal-tea.png');
        break;

      case 'jumbo oats':
        icon = require('../../../assets/img/products/jumbo-oats.png');
        break;

      case 'milk':
        icon = require('../../../assets/img/products/milk.png');
        break;

      case 'organic meat patties':
        icon = require('../../../assets/img/products/organic-meat-patties.png');
        break;

      case 'organic coffee':
        icon = require('../../../assets/img/products/organic-coffee.png');
        break;

      case 'parma ham':
        icon = require('../../../assets/img/products/parma-ham.png');
        break;

      case 'porridge oats':
        icon = require('../../../assets/img/products/porridge-oats.png');
        break;

      case 'pumpkin seed bun':
        icon = require('../../../assets/img/products/pumpkin-seed-bun.png');
        break;

      case 'red currants':
        icon = require('../../../assets/img/products/red-currants.png');
        break;

      case 'rice flakes':
        icon = require('../../../assets/img/products/rice-flakes.png');
        break;

      case 'sausages':
        icon = require('../../../assets/img/products/sausages.png');
        break;

      case 'seedless watermelon':
        icon = require('../../../assets/img/products/seedless-watermelon.png');
        break;

      case 'sesame seed bagel':
        icon = require('../../../assets/img/products/sesame-seed-bagel.png');
        break;

      case 'sunflower seed loaf':
        icon = require('../../../assets/img/products/sunflower-seed-loaf.png');
        break;

      case 'wholebean coffee':
        icon = require('../../../assets/img/products/wholebean-coffee.png');
        break;

      default:
    }

    return icon;
  };

  render() {
    const {categories} = this.props;

    if (categories.length === 0) {
      return null;
    }

    return (
      <View
        style={{
          alignItems: 'center',
        }}>
        <FlatList
          data={this.props.products._embedded.productList}
          keyExtractor={(item, index) => item.id}
          renderItem={({item}) => {
            return (
              <View
                style={{
                  alignItems: 'center',
                  flexDirection: 'row',
                  paddingBottom: 24,
                }}>
                <Image source={this.getProductIcon(item.name)} />
                <View style={{paddingLeft: 24}} />
                <View style={{alignItems: 'flex-start'}}>
                  <Text style={{fontWeight: 'bold'}}>{item.name}</Text>
                  <Text style={{paddingBottom: 8}}>
                    &euro; {item.price.toFixed(2)}
                  </Text>
                  <Button onPress={() => this.addToCart(item.id)} title="add" />
                </View>
              </View>
            );
          }}
        />
      </View>
    );
  }
}

Products.propTypes = {
  category: PropTypes.object.isRequired,
  categories: PropTypes.array.isRequired,
  dispatch: PropTypes.func.isRequired,
};
