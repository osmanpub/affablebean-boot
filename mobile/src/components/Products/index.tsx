import React from 'react';
import {
  ActivityIndicator,
  Button,
  FlatList,
  Image,
  Text,
  View,
} from 'react-native';
import {connect, useDispatch} from 'react-redux';
import {getId, getProductIcon} from '../../helpers/utils';
import {Cart} from '../../interfaces/cart';
import {
  Category,
  CategoryState,
  ProductState,
} from '../../interfaces/categories';
import {id} from '../../interfaces/id';
import {addProductToCart, updateProductInCart} from '../../net/cart';
import {RootState} from '../../redux';
import {clearPurchase} from '../../redux/purchase';

type Props = {
  cart: Cart;
  categories: Array<CategoryState>;
  category: Category;
  clearPurchase: Function;
  products: Array<ProductState>;
};

function Products(props: Props) {
  const {cart, categories, clearPurchase, products} = props;
  const dispatch = useDispatch();

  const addToCart = (id: id) => {
    const update = cart.items.filter(item => getId(item.product) === id);

    clearPurchase();
    dispatch(
      update.length > 0
        ? updateProductInCart(id, update[0].quantity + 1)
        : addProductToCart(id.toString()),
    );
  };

  if (categories.length === 0) {
    return null;
  }

  return cart.isFetching ? (
    <ActivityIndicator size="large" color="blue" />
  ) : (
    <View
      style={{
        alignItems: 'center',
      }}>
      <FlatList
        data={products}
        keyExtractor={item => getId(item).toString()}
        renderItem={({item}) => {
          return (
            <View
              style={{
                alignItems: 'center',
                flexDirection: 'row',
                paddingBottom: 24,
              }}>
              <Image source={getProductIcon(item.name)} />
              <View style={{paddingLeft: 24}} />
              <View style={{alignItems: 'flex-start'}}>
                <Text style={{fontWeight: 'bold'}}>{item.name}</Text>
                <Text style={{paddingBottom: 8}}>
                  &euro; {item.price.toFixed(2)}
                </Text>
                <Button onPress={() => addToCart(getId(item))} title="add" />
              </View>
            </View>
          );
        }}
      />
    </View>
  );
}

const mapStateToProps = (state: RootState) => ({
  cart: state.cart,
});

const mapDispatchToProps = {
  clearPurchase,
};

export default connect(mapStateToProps, mapDispatchToProps)(Products);
