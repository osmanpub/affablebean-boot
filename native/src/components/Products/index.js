import React, { Component } from "react";
import PropTypes from "prop-types";
import { FlatList, Image, StyleSheet, Text, View } from "react-native";
import { addProductToCart } from "../../net/cart";
import { clearPurchase } from "../../actions";

export default class Products extends Component {
  addToCart = id => {
    const { dispatch } = this.props;
    dispatch(clearPurchase());
    dispatch(addProductToCart(id));
  };

  getProductIcon = name => {
    var icon = "";

    switch (name) {
      case "chocolate cookies":
        icon = require("./products/chocolate-cookies.png");
        break;

      case "pumpkin seed bun":
        icon = require("./products/pumpkin-seed-bun.png");
        break;

      case "sesame seed bagel":
        icon = require("./products/sesame-seed-bagel.png");
        break;

      case "sunflower seed loaf":
        icon = require("./products/sunflower-seed-loaf.png");
        break;

      default:
    }

    return icon;
  };

  render() {
    const { categories } = this.props;

    if (categories.length === 0) {
      return null;
    }

    const styles = StyleSheet.create({
      productImage: {
        padding: 1
      }
    });

    return (
      <View
        style={{
          alignItems: "center",
          flex: 1
        }}
      >
        <FlatList
          data={this.props.products._embedded.productList}
          keyExtractor={(item, index) => item._links.self.href}
          renderItem={({ item }) => {
            return (
              <View>
                <Text>{item.name}</Text>
                <Image
                  source={this.getProductIcon(item.name)}
                  style={styles.productImage}
                />
                <Text />
                <Text />
                <Text />
                <Text />
                <Text />
                <Text />
              </View>
            );
          }}
        />
      </View>
    );
  }
}

Products.propTypes = {
  category: PropTypes.object.isRequired,
  categories: PropTypes.array.isRequired,
  dispatch: PropTypes.func.isRequired
};
