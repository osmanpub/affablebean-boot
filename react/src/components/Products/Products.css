.categoryButton {
  margin: 15px 22px;
  padding: 13px;
  display: block;
  background-color: #d3ede8;
  border-radius: 4px;
}

.categoryText {
  line-height: 25px;
  font-size: x-large;
}
