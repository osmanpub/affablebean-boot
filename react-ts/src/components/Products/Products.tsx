import React, { Component } from "react";
import { Link } from "react-router-dom";
import "./Products.css";
import { addProductToCart } from "../../net/cart";
import { clearPurchase } from "../../actions";
import {
  CategoryTitle,
  ProductsLeft,
  ProductsRight,
  ProductsTable,
  SelectedCategory
} from "./Products.styles";
import { CategoryProductsProps } from "../../interfaces/props";
import { Product } from "../../interfaces/categories";

export class Products extends Component<CategoryProductsProps> {
  addToCart = (id: number) => {
    const { dispatch } = this.props;
    dispatch(clearPurchase());
    dispatch(addProductToCart(id.toString()));
  };

  render() {
    const { categories } = this.props;

    if (categories.length === 0) {
      return null;
    }

    const selectedCategory = this.props.category;

    const sidePanel = categories.map(category => {
      const key = category.id;
      const name = category.name;

      if (name === selectedCategory.name) {
        return (
          <SelectedCategory key={key}>
            <span className="categoryText">{name}</span>
          </SelectedCategory>
        );
      } else {
        return (
          <Link key={key} to={`/category/${category.id}`}>
            <span className="categoryButton">
              <span className="categoryText">{name}</span>
            </span>
          </Link>
        );
      }
    });

    const products = this.props.products._embedded.productList.map(
      (product: Product, index: number) => {
        const name = product.name;
        const rowCol = index % 2 === 0 ? "white" : "lightBlue";

        return (
          <tr key={product.id} className={`${rowCol}`}>
            <td>
              <img src={`/static/img/products/${name}.png`} alt="{name}" />
            </td>

            <td>
              {name}
              <br />
              <span className="smallText">{product.description}</span>
            </td>

            <td>
              &euro;&nbsp;
              {product.price.toFixed(2)}
            </td>

            <td>&nbsp;&nbsp;</td>

            <td>
              <button
                className="`btn btn-primary btn-sm`"
                onClick={this.addToCart.bind(this, product.id)}
              >
                add
              </button>
            </td>

            <td>&nbsp;&nbsp;</td>
          </tr>
        );
      }
    );

    return (
      <div>
        <ProductsLeft>{sidePanel}</ProductsLeft>
        <ProductsRight>
          <CategoryTitle>{selectedCategory.name}</CategoryTitle>
          <ProductsTable>
            <tbody>{products}</tbody>
          </ProductsTable>
        </ProductsRight>
      </div>
    );
  }
}
