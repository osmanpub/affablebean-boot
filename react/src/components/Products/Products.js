import React, { Component } from "react";
import { Link } from "react-router-dom";
import "./Products.css";
import { addProductToCart } from "../../rest/cart";
import {
  CategoryTitle,
  ProductsLeft,
  ProductsRight,
  ProductsTable,
  SelectedCategory
} from "./Products.styles";

export class Products extends Component {
  constructor(props) {
    super(props);
    this.addToCart = this.addToCart.bind(this);
  }

  addToCart(id) {
    const { dispatch } = this.props;
    dispatch(addProductToCart(id));
  }

  render() {
    const categories = this.props.categories;

    if (categories.length === 0) {
      return null;
    }

    const selectedCategory = this.props.category;

    const sidePanel = categories.map(category => {
      const key = category._links.self.href;
      const name = category.name;

      if (name === selectedCategory.name) {
        return (
          <SelectedCategory key={key}>
            <span className="categoryText">{name}</span>
          </SelectedCategory>
        );
      } else {
        return (
          <Link key={key} to={"/category/" + category.id}>
            <span className="categoryButton">
              <span className="categoryText">{name}</span>
            </span>
          </Link>
        );
      }
    });

    const products = this.props.products._embedded.productList.map(
      (product, index) => {
        const name = product.name;

        return (
          <tr
            key={product._links.self.href}
            className="{index % 2 === 0 ? 'white' : 'lightBlue'}"
          >
            <td>
              <img
                src={"/static/img/products/" + name + ".png"}
                alt="{{product.name}"
              />
            </td>

            <td>
              <span>{name}</span>
              <br />
              <span className="smallText">{product.description}</span>
            </td>

            <td>
              &euro;&nbsp;
              {product.price}
            </td>

            <td>&nbsp;&nbsp;</td>

            <td>
              <button
                className="btn btn-primary btn-sm"
                onClick={this.addToCart.bind(this, product.id)}
              >
                <span>add</span>
              </button>
            </td>

            <td>&nbsp;&nbsp;</td>
          </tr>
        );
      }
    );

    return (
      <div>
        <ProductsLeft>{sidePanel}</ProductsLeft>
        <ProductsRight>
          <CategoryTitle>{selectedCategory.name}</CategoryTitle>
          <ProductsTable>
            <tbody>{products}</tbody>
          </ProductsTable>
        </ProductsRight>
      </div>
    );
  }
}
