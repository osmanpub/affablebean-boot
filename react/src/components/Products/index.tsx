import React from "react";
import { connect, useDispatch } from "react-redux";
import { Link } from "react-router-dom";
import { getId } from "../../helpers/utils";
import { Cart } from "../../interfaces/cart";
import {
  Category,
  CategoryState,
  ProductState
} from "../../interfaces/categories";
import { id } from "../../interfaces/id";
import { addProductToCart, updateProductInCart } from "../../net/cart";
import { RootState } from "../../redux";
import { clearPurchase } from "../../redux/purchase";
import "./Products.css";
import {
  CategoryTitle,
  ProductsLeft,
  ProductsRight,
  ProductsTable,
  SelectedCategory
} from "./Products.styles";

type Props = {
  cart: Cart;
  categories: Array<CategoryState>;
  category: Category;
  clearPurchase: Function;
  products: Array<ProductState>;
};

function Products(props: Props) {
  const { cart, categories, clearPurchase, products } = props;
  const dispatch = useDispatch();

  const addToCart = (id: id) => {
    const update = cart.items.filter(item => getId(item.product) === id);

    clearPurchase();
    dispatch(
      update.length > 0
        ? updateProductInCart(id, update[0].quantity + 1)
        : addProductToCart(id.toString())
    );
  };

  if (categories.length === 0) {
    return null;
  }

  const selectedCategory = props.category;

  const sidePanel = categories.map((category: CategoryState) => {
    const id = getId(category);
    const name = category.name;

    if (name === selectedCategory.name) {
      return (
        <SelectedCategory key={id}>
          <span className="categoryText">{name}</span>
        </SelectedCategory>
      );
    } else {
      return (
        <Link key={id} to={`/category/${id}`}>
          <span className="categoryButton">
            <span className="categoryText" data-cy={`category-${name}`}>
              {name}
            </span>
          </span>
        </Link>
      );
    }
  });

  const productsList = products.map((product: ProductState, index: number) => {
    const id = getId(product);
    const name = product.name;
    const rowCol = index % 2 === 0 ? "white" : "lightBlue";

    return (
      <tr key={id} className={`${rowCol}`} data-cy={`product-${name}`}>
        <td>
          <img src={`/static/img/products/${name}.png`} alt="{name}" />
        </td>

        <td>
          {name}
          <br />
          <span className="smallText">{product.description}</span>
        </td>

        <td>
          &euro;&nbsp;
          {product.price.toFixed(2)}
        </td>

        <td>&nbsp;&nbsp;</td>

        <td>
          <button
            className="btn btn-primary btn-sm"
            onClick={() => addToCart(id)}
          >
            add
          </button>
        </td>

        <td>&nbsp;&nbsp;</td>
      </tr>
    );
  });

  return (
    <div>
      <ProductsLeft>{sidePanel}</ProductsLeft>
      <ProductsRight>
        <CategoryTitle>{selectedCategory.name}</CategoryTitle>
        <ProductsTable>
          <tbody>{productsList}</tbody>
        </ProductsTable>
      </ProductsRight>
    </div>
  );
}

const mapStateToProps = (state: RootState) => ({
  cart: state.cart
});

const mapDispatchToProps = {
  clearPurchase
};

export default connect(mapStateToProps, mapDispatchToProps)(Products);
