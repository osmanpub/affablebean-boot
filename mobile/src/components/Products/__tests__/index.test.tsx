import React from 'react';
import {Provider} from 'react-redux';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import Products from '..';

const mockStore = configureStore([]);

const initialState = {
  cart: {
    didInvalidate: false,
    isFetching: false,
    items: [],
    numberOfItems: 0,
    subtotal: 0,
  },
};

const store = mockStore(initialState);

const category = {
  categories: [
    {
      id: 3,
      name: 'bakery',
      _links: {
        self: {href: 'http://localhost:8080/api/categories/3'},
        categories: {href: 'http://localhost:8080/api/categories'},
      },
    },
    {
      id: 5,
      name: 'cereals',
      _links: {
        self: {href: 'http://localhost:8080/api/categories/5'},
        categories: {href: 'http://localhost:8080/api/categories'},
      },
    },
    {
      id: 1,
      name: 'dairy',
      _links: {
        self: {href: 'http://localhost:8080/api/categories/1'},
        categories: {href: 'http://localhost:8080/api/categories'},
      },
    },
    {
      id: 6,
      name: 'drinks',
      _links: {
        self: {href: 'http://localhost:8080/api/categories/6'},
        categories: {href: 'http://localhost:8080/api/categories'},
      },
    },
    {
      id: 4,
      name: 'fruit & veg',
      _links: {
        self: {href: 'http://localhost:8080/api/categories/4'},
        categories: {href: 'http://localhost:8080/api/categories'},
      },
    },
    {
      id: 2,
      name: 'meats',
      _links: {
        self: {href: 'http://localhost:8080/api/categories/2'},
        categories: {href: 'http://localhost:8080/api/categories'},
      },
    },
  ],
  category: {
    id: 1,
    name: 'dairy',
    _links: {
      self: {href: 'http://localhost:8080/api/categories/1'},
      categories: {href: 'http://localhost:8080/api/categories'},
    },
  },
  products: [
    // node
    {
      id: 1,
      description: 'semi skimmed (1L)',
      name: 'milk',
      price: 1.7,
      category: {id: 1, name: 'dairy'},
      _links: {
        self: {href: 'http://localhost:8080/api/products/1'},
        products: {href: 'http://localhost:8080/api/products'},
      },
    },
    {
      id: 2,
      description: 'mild cheddar (330g)',
      name: 'cheese',
      price: 2.39,
      category: {id: 1, name: 'dairy'},
      _links: {
        self: {href: 'http://localhost:8080/api/products/2'},
        products: {href: 'http://localhost:8080/api/products'},
      },
    },
    {
      id: 3,
      description: 'unsalted (250g)',
      name: 'butter',
      price: 1.09,
      category: {id: 1, name: 'dairy'},
      _links: {
        self: {href: 'http://localhost:8080/api/products/3'},
        products: {href: 'http://localhost:8080/api/products'},
      },
    },
    {
      id: 4,
      description: 'medium-sized (6 eggs)',
      name: 'free range eggs',
      price: 1.76,
      category: {id: 1, name: 'dairy'},
      _links: {
        self: {href: 'http://localhost:8080/api/products/4'},
        products: {href: 'http://localhost:8080/api/products'},
      },
    },
  ],
  // products: { // java
  //   _embedded: {
  //     productList: [
  //       {
  //         id: 1,
  //         description: "semi skimmed (1L)",
  //         name: "milk",
  //         price: 1.7,
  //         category: { id: 1, name: "dairy" },
  //         _links: {
  //           self: { href: "http://localhost:8080/api/products/1" },
  //           products: { href: "http://localhost:8080/api/products" }
  //         }
  //       },
  //       {
  //         id: 2,
  //         description: "mild cheddar (330g)",
  //         name: "cheese",
  //         price: 2.39,
  //         category: { id: 1, name: "dairy" },
  //         _links: {
  //           self: { href: "http://localhost:8080/api/products/2" },
  //           products: { href: "http://localhost:8080/api/products" }
  //         }
  //       },
  //       {
  //         id: 3,
  //         description: "unsalted (250g)",
  //         name: "butter",
  //         price: 1.09,
  //         category: { id: 1, name: "dairy" },
  //         _links: {
  //           self: { href: "http://localhost:8080/api/products/3" },
  //           products: { href: "http://localhost:8080/api/products" }
  //         }
  //       },
  //       {
  //         id: 4,
  //         description: "medium-sized (6 eggs)",
  //         name: "free range eggs",
  //         price: 1.76,
  //         category: { id: 1, name: "dairy" },
  //         _links: {
  //           self: { href: "http://localhost:8080/api/products/4" },
  //           products: { href: "http://localhost:8080/api/products" }
  //         }
  //       }
  //     ]
  //   },
  //   _links: {
  //     categoryProducts: { href: "http://localhost:8080/api/products" }
  //   }
  // }
};

describe('<Products />', () => {
  it('renders correctly', () => {
    const component = renderer
      .create(
        <Provider store={store}>
          <Products {...category} />
        </Provider>,
      )
      .toJSON();
    expect(component).toMatchSnapshot();
  });
});
