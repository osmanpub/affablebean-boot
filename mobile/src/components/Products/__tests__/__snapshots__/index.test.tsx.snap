// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Products /> renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
    }
  }
>
  <RCTScrollView
    data={
      Array [
        Object {
          "_links": Object {
            "products": Object {
              "href": "http://localhost:8080/api/products",
            },
            "self": Object {
              "href": "http://localhost:8080/api/products/1",
            },
          },
          "category": Object {
            "id": 1,
            "name": "dairy",
          },
          "description": "semi skimmed (1L)",
          "id": 1,
          "name": "milk",
          "price": 1.7,
        },
        Object {
          "_links": Object {
            "products": Object {
              "href": "http://localhost:8080/api/products",
            },
            "self": Object {
              "href": "http://localhost:8080/api/products/2",
            },
          },
          "category": Object {
            "id": 1,
            "name": "dairy",
          },
          "description": "mild cheddar (330g)",
          "id": 2,
          "name": "cheese",
          "price": 2.39,
        },
        Object {
          "_links": Object {
            "products": Object {
              "href": "http://localhost:8080/api/products",
            },
            "self": Object {
              "href": "http://localhost:8080/api/products/3",
            },
          },
          "category": Object {
            "id": 1,
            "name": "dairy",
          },
          "description": "unsalted (250g)",
          "id": 3,
          "name": "butter",
          "price": 1.09,
        },
        Object {
          "_links": Object {
            "products": Object {
              "href": "http://localhost:8080/api/products",
            },
            "self": Object {
              "href": "http://localhost:8080/api/products/4",
            },
          },
          "category": Object {
            "id": 1,
            "name": "dairy",
          },
          "description": "medium-sized (6 eggs)",
          "id": 4,
          "name": "free range eggs",
          "price": 1.76,
        },
      ]
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "paddingBottom": 24,
            }
          }
        >
          <Image
            source={
              Object {
                "testUri": "../../../assets/img/products/milk.png",
              }
            }
          />
          <View
            style={
              Object {
                "paddingLeft": 24,
              }
            }
          />
          <View
            style={
              Object {
                "alignItems": "flex-start",
              }
            }
          >
            <Text
              style={
                Object {
                  "fontWeight": "bold",
                }
              }
            >
              milk
            </Text>
            <Text
              style={
                Object {
                  "paddingBottom": 8,
                }
              }
            >
              € 
              1.70
            </Text>
            <View
              accessibilityRole="button"
              accessibilityStates={Array []}
              accessible={true}
              focusable={true}
              isTVSelectable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <View
                style={
                  Array [
                    Object {},
                  ]
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#007AFF",
                        "fontSize": 18,
                        "padding": 8,
                        "textAlign": "center",
                      },
                    ]
                  }
                >
                  add
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "paddingBottom": 24,
            }
          }
        >
          <Image
            source={
              Object {
                "testUri": "../../../assets/img/products/cheese.png",
              }
            }
          />
          <View
            style={
              Object {
                "paddingLeft": 24,
              }
            }
          />
          <View
            style={
              Object {
                "alignItems": "flex-start",
              }
            }
          >
            <Text
              style={
                Object {
                  "fontWeight": "bold",
                }
              }
            >
              cheese
            </Text>
            <Text
              style={
                Object {
                  "paddingBottom": 8,
                }
              }
            >
              € 
              2.39
            </Text>
            <View
              accessibilityRole="button"
              accessibilityStates={Array []}
              accessible={true}
              focusable={true}
              isTVSelectable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <View
                style={
                  Array [
                    Object {},
                  ]
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#007AFF",
                        "fontSize": 18,
                        "padding": 8,
                        "textAlign": "center",
                      },
                    ]
                  }
                >
                  add
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "paddingBottom": 24,
            }
          }
        >
          <Image
            source={
              Object {
                "testUri": "../../../assets/img/products/butter.png",
              }
            }
          />
          <View
            style={
              Object {
                "paddingLeft": 24,
              }
            }
          />
          <View
            style={
              Object {
                "alignItems": "flex-start",
              }
            }
          >
            <Text
              style={
                Object {
                  "fontWeight": "bold",
                }
              }
            >
              butter
            </Text>
            <Text
              style={
                Object {
                  "paddingBottom": 8,
                }
              }
            >
              € 
              1.09
            </Text>
            <View
              accessibilityRole="button"
              accessibilityStates={Array []}
              accessible={true}
              focusable={true}
              isTVSelectable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <View
                style={
                  Array [
                    Object {},
                  ]
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#007AFF",
                        "fontSize": 18,
                        "padding": 8,
                        "textAlign": "center",
                      },
                    ]
                  }
                >
                  add
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "paddingBottom": 24,
            }
          }
        >
          <Image
            source={
              Object {
                "testUri": "../../../assets/img/products/free-range-eggs.png",
              }
            }
          />
          <View
            style={
              Object {
                "paddingLeft": 24,
              }
            }
          />
          <View
            style={
              Object {
                "alignItems": "flex-start",
              }
            }
          >
            <Text
              style={
                Object {
                  "fontWeight": "bold",
                }
              }
            >
              free range eggs
            </Text>
            <Text
              style={
                Object {
                  "paddingBottom": 8,
                }
              }
            >
              € 
              1.76
            </Text>
            <View
              accessibilityRole="button"
              accessibilityStates={Array []}
              accessible={true}
              focusable={true}
              isTVSelectable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <View
                style={
                  Array [
                    Object {},
                  ]
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#007AFF",
                        "fontSize": 18,
                        "padding": 8,
                        "textAlign": "center",
                      },
                    ]
                  }
                >
                  add
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
